### ISSUE TITLE
BackstopException causes backstop process to hang indefinitely

### ISSUE THREAD

**Issue by laubsauger** [garris/BackstopJS#549]
If we encounter a BackstopException for any reason, the whole process just hangs after throwing the exception. Makes it difficult to set it up for CI. 

**Setup**
clean environment, backstop default scenario

**Reproduction**
I've added an evaluation with an undefined variable to `onReady.js`
````javascript
module.exports = function (chromy, scenario, vp) {
  console.log('SCENARIO > ' + scenario.label);
  require('./clickAndHoverHelper')(chromy, scenario);
  // add more ready handlers here...

    chromy.evaluate(() => {
      IAmTotallyUndefined // will cause BackstopException -> undefined variable
    });
};
````
Error output
````bash
COMMAND | Command `reference` ended with an error after [1.416s]
COMMAND | BackstopException: BackstopJS Homepage on undefined: Error: An error has occurred evaluating the script in the browser.ReferenceError: IAmTotallyUndefined is not defined
    at result (<anonymous>:3:7)
    at <anonymous>:4:7
````
afterwards hanging forever until process is exited manually

Encountering TimeoutExceptions or other Exceptions correctly exits. 
Any idea where to start?

---
**edit**: for now i went with a bare bones approach of calling `process.exit(1)` in the promise.catch block after logging the error. Seems like a bad solution though. chrome instances are not properly shutdown.

**Comment by garris** [comment]
Thank you for looking into this.  Yes, this a really annoying behavior!   It should really respond with a failed test icon — but not break existing tests still in flight.   

I think this should not be too hard to fix — it would be great if you could post your current PR.  We could improve it here.

**Comment by mirovelk** [comment]
Same problem here. Wouldn't it be enough to continue as usual even when chromy script crashes? There's no need to throw exceptions or fail tests. Tests should fail naturally after screenshot comparison (if user interaction simulation failed, application will be in a different state than expected). Maybe just log that the script failed so we can see that there's something wrong...

**Comment by garris** [comment]
@mirovelk totally agree -- this is the most annoying thing about backstopjs!  I would merge any PR that fixes this so fast it'd make your socks fly off!

**Comment by garris** [comment]
Ok backstoppers!   This problem is about to go away.   I have a fix — need a few days to validate it.   Will update soon!

Cheers.

**Comment by garris** [comment]
fixed by c787c5b
